onnxruntime/core/providers/cann/npu_data_transfer.cc

```diff
@@ -86,17 +92,23 @@ common::Status NPUDataTransfer::CopyTensorAsync(const Tensor& src, Tensor& dst,
                                               ACL_MEMCPY_DEVICE_TO_DEVICE,
                                               static_cast<aclrtStream>(stream.GetHandle())));
       }
+    } else {
+      // copy from pinned or CPU memory to NPU, this is non-blocking
+      CANN_RETURN_IF_ERROR(aclrtMemcpyAsync(dst_data,
+                                            bytes,
+                                            src_data,
+                                            bytes,
+                                            ACL_MEMCPY_HOST_TO_DEVICE,
+                                            static_cast<aclrtStream>(stream.GetHandle())));
     }
-  } else if (src_device.Type() == OrtDevice::NPU) {
-    if (dst_device.Type() == OrtDevice::CPU) {
```

Comment:
this `if` was unnecessary as CPU in the old code covered CPU and host accessible, so there are no other destination options as we did npu to npu earlier.